### DISCUSSION TITLE
Timeout when cloning Git repository

### DISCUSSION THREAD

**Discussion by Babbleshack** [argoproj/argo-cd#21054] [Q&A] [answered]
I am experiencing a timeout issue while cloning our Git repository from the app-set controller and repo-server. The error message I receive is as follows:

```
error generating params from git: error getting directories from repo: error retrieving Git Directories: rpc error: code = Internal desc = unable to checkout git repo git@bitbucket.org:XXXXXX/XXXXXX.git with revision XXXXXXXXXXXXXXXX: failed to initialize repository resources: rpc error: code = Internal desc = Failed to fetch default: `git fetch origin --tags --force --prune` failed timeout after 1m30s
```

For context, cloning the repository locally takes around 95 seconds.

To address the timeout I tried setting `reposerver.git.request.timeout" = "3m"`. 

I verified these setting were correctly configured by investigating environment variables on the deployed repo-server pods:

```bash
$ kubectl exec deployments/argo-cd-argocd-repo-server -c repo-server -- env | grep TIMEOUT
ARGOCD_GIT_REQUEST_TIMEOUT=3m
ARGOCD_RECONCILIATION_TIMEOUT=180s
```

Despite these changes, the logs still indicate timeout messages suggesting that the timeout remains at 1m30s.

Are there additional configs I must change to increase the git clone timeout?

**Comment by rumstead** [comment, marked as answer]
The timeout for a git fetch is controlled by `ARGOCD_EXEC_TIMEOUT` because the [git fetch](https://github.com/argoproj/argo-cd/blob/cdec7e28686a0819f8445f08d587999c1d5d9602/util/git/client.go#L355) is [execed](https://github.com/argoproj/argo-cd/blob/cdec7e28686a0819f8445f08d587999c1d5d9602/util/git/client.go#L890) and the git client isn't used. It isn't clear in the docs that this is the case. Would you mind creating an issue or even submitting a PR to help update the docs?

**Reply by Babbleshack** [reply]
Awesome thanks for this. I will do a PR soon.
